http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/auth/sam/SamAuthProvider.scala

```diff
@@ -134,23 +131,20 @@ class SamAuthProvider(val config: Config, serviceAccountProvider: ServiceAccount
     * @param clusters All non-deleted clusters from the database
     * @return         Filtered list of clusters that the user is allowed to see
     */
-  override def filterUserVisibleClusters(userInfo: UserInfo, clusters: List[(GoogleProject, ClusterName)])(implicit executionContext: ExecutionContext): Future[List[(GoogleProject, ClusterName)]] = {
+  override def filterUserVisibleClusters(userInfo: UserInfo, clusters: List[(GoogleProject, ClusterInternalId)])(implicit ev: ApplicativeAsk[F, TraceId]): F[List[(GoogleProject, ClusterInternalId)]] = {
+    val authHeader = Authorization(Credentials.Token(AuthScheme.Bearer, userInfo.accessToken.token))
     for {
-      owningProjects <- retryableFutureToFuture(retryUntilSuccessOrTimeout(shouldInvalidateSamCacheAndRetry, s"SamAuthProvider.filterUserVisibleClusters.owningProjects call failed")(samRetryInterval, samRetryTimeout) { () =>
-        Future(blocking(samClient.listOwningProjects(userInfo)).toSet)
-      })
-      createdClusters <- retryableFutureToFuture(retryUntilSuccessOrTimeout(shouldInvalidateSamCacheAndRetry, s"SamAuthProvider.filterUserVisibleClusters.createdClusters call failed")(samRetryInterval, samRetryTimeout) { () =>
-        Future(blocking(samClient.listCreatedClusters(userInfo)).toSet)
-      })
-    } yield {
-      clusters.filter { case (project, name) =>
-        owningProjects.contains(project) || createdClusters.contains((project, name))
-      }
+      projectPolicies <- samDao.getResourcePolicies[SamProjectPolicy](authHeader, ResourceTypeName.BillingProject)
+      owningProjects = projectPolicies.collect {case x if(x.accessPolicyName == AccessPolicyName.Owner) => x.googleProject}
+      clusterPolicies <- samDao.getResourcePolicies[SamNotebookClusterPolicy](authHeader, ResourceTypeName.NotebookCluster)
+      createdPolicies = clusterPolicies.filter(_.accessPolicyName == AccessPolicyName.Creator)
```

Comment:
Should we `.collect` here instead of `.filter` so we can do another `.contains` below instead of `.exists`?